src/nvim/ops.c

```diff
@@ -3644,6 +3644,10 @@ int do_join(size_t count,
    */
   for (t = (linenr_T)count - 1;; t--) {
     cend -= currsize;
+	if(cend == NULL || curr == NULL){
```

Comment:
I'll do that, but I don't see where `curr` depends on `count`. Additionally, we keep subtracting from `cend`, shouldn't we still check for NULL as we subtract?